src/adagrams.js

```diff
@@ -1,15 +1,79 @@
+import { SCORE_CHART, LETTER_POOL } from "constants";
+
+//creates letter pool
+export const createLetterPool = () => {
+  let letterPool = [];
+  for (const [key, value] of Object.entries(LETTER_POOL)) {
+    for (let i = 0; i < value; i++) {
+      letterPool.push(key);
+    }
+  }
+  return letterPool;
+};
+
+//shuffles letter pool
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let letterPool = createLetterPool();
+  let letterPoolDeepCopy = JSON.parse(JSON.stringify(letterPool));
+  function shuffleArray(letterPoolDeepCopy) {
+    for (let i = 0; i < letterPoolDeepCopy; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    }
+  }
+  let letterBank = letterPoolDeepCopy.slice(0, 10);
+  return letterBank;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  for (let letters = 0; letters < input.length; letters++) {
+    let char = input[letters];
+    if (lettersInHand.includes(char) === false) {
+      return false;
+    } else if (lettersInHand.includes(char)) {
```

Comment:
Either a letter is included or not, so this could be an else clause rather than else if
```suggestion
    } else {
```